// gitbook-plugin-tbfed-pagefooter/index.js
var moment = require('moment');
module.exports = {
  book: {
    assets: './assets',
    css: [
      'footer.css'
    ],
  },
  hooks: {
    'page:before': function(page) {
      var _label = '更新时间: ',
          _format = 'YYYY-MM-DD',
          _copy = 'powered by sysdsoft'
      if(this.options.pluginsConfig['tbfed-pagefooter']) {
        _label = this.options.pluginsConfig['tbfed-pagefooter']['modify_label'] || _label;
        _format = this.options.pluginsConfig['tbfed-pagefooter']['modify_format'] || _format;

        var _c = this.options.pluginsConfig['tbfed-pagefooter']['copyright'];
        _copy = _c ? _c + ' all right reserved，' + _copy : _copy;
      }
      var _copy = '<span class="copyright">'+_copy+'</span>'
	  
      var str = '\n\n<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/gitalk@1/dist/gitalk.css">'+
      '\n<script src="https://cdn.jsdelivr.net/npm/gitalk@1/dist/gitalk.min.js"></script>'+
      '\n<div id="gitalk-container"></div>'+
	  '\n<script src="http://cd.sysdsoft.cn:4000/md5.min.js"></script>'+
      '\n<script src="http://cd.sysdsoft.cn:4000/gitalk-config.js"></script>'+
	  ' \n\n<footer class="page-footer">' + _copy +
        '<span class="footer-modification">' +
        _label +
        '\n{{file.mtime | date("' + _format +
        '")}}\n</span></footer>';
		
		
      page.content = page.content + str;
	  
      return page;

    }
  },
  filters: {
    date: function(d, format) {
      return moment(d).format(format)
    }
  }
};
